include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -49,16 +44,31 @@
 #    define _ONEDPL_PREDEFINED_POLICIES 1
 #endif
 
-#if ONEDPL_USE_TBB_BACKEND || (!defined(ONEDPL_USE_TBB_BACKEND) && !ONEDPL_USE_OPENMP_BACKEND)
-#    define _ONEDPL_PAR_BACKEND_TBB 1
+// Check availability of parallel backends
+#if __has_include(<tbb/tbb.h>)
+#    define _ONEDPL_TBB_AVAILABLE 1
+#endif
+#if ONEDPL_USE_TBB_BACKEND && !_ONEDPL_TBB_AVAILABLE
+#    error "Parallel execution policies with oneTBB or Intel(R) TBB support are enabled, but the library is not found"
 #endif
 
-#if ONEDPL_USE_OPENMP_BACKEND || (!defined(ONEDPL_USE_OPENMP_BACKEND) && defined(_OPENMP))
-#    define _ONEDPL_PAR_BACKEND_OPENMP 1
+#if defined(_OPENMP) && __has_include(<omp.h>)
+#    define _ONEDPL_OPENMP_AVAILABLE 1
+#endif
+// During compilation for a device _OPENMP may not be set, so avoid throwing an error if
+// __SYCL_DEVICE_ONLY__ found
+#if ONEDPL_USE_OPENMP_BACKEND && !_ONEDPL_OPENMP_AVAILABLE && !defined(__SYCL_DEVICE_ONLY__)
+#    error "Parallel execution policies with OpenMP* support are enabled, \
+        but OpenMP headers are not found or the compiler does not support OpenMP"
 #endif
 
-#if !_ONEDPL_PAR_BACKEND_TBB && !_ONEDPL_PAR_BACKEND_OPENMP
-#    define _ONEDPL_PAR_BACKEND_SERIAL 1
+#if (defined(CL_SYCL_LANGUAGE_VERSION) || defined(SYCL_LANGUAGE_VERSION)) &&                                           \
+    (__has_include(<sycl/sycl.hpp>) || __has_include(<CL/sycl.hpp>))
```

Comment:
```suggestion
#if (defined(SYCL_LANGUAGE_VERSION) || defined(CL_SYCL_LANGUAGE_VERSION)) &&                                           \
    (__has_include(<sycl/sycl.hpp>) || __has_include(<CL/sycl.hpp>))
```

For consistency with the second line.